Node Pre GYP Install response error and build error in Ubuntu 21.04 / NodeJS 16.9.1 / Python 3.9.5

When attempting to install using the latest version of symbology via Node 16.9.1 it fails to download the prebuilt node build file and also fails to fallback compile using node-gyp.

I have tested on multiple machines in different networks and different operating systems. I have even checked via wget if https://github.com/jshor/symbology/releases/download/3.0.6/symbology_linux_x64_node-v93.tar.gz is accessible by the system and it can download it no problem however, during the build process it errors.

The node-gyp fallback build also fails with the error ../src/binding/main.cpp:10:10: fatal error: nan.h: No such file or directory
   10 | #include <nan.h>

OS: Ubuntu 21.04
Node: 16.9.1
Node-GYP: node-gyp@7.1.2

`> symbology@3.0.6 binary:install
> node-pre-gyp install --fallback-to-build

node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@1.0.5
node-pre-gyp info using node@16.9.1 | linux | x64
node-pre-gyp info check checked for "/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin/symbology.node" (not found)
node-pre-gyp http GET https://github.com/jshor/symbology/releases/download/3.0.6/symbology_linux_x64_node-v93.tar.gz
node-pre-gyp ERR! install response status 403 Forbidden on https://github.com/jshor/symbology/releases/download/3.0.6/symbology_linux_x64_node-v93.tar.gz 
node-pre-gyp WARN Pre-built binaries not installable for symbology@3.0.6 and node@16.9.1 (node-v93 ABI, glibc) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error response status 403 Forbidden on https://github.com/jshor/symbology/releases/download/3.0.6/symbology_linux_x64_node-v93.tar.gz 
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.9.1 | linux | x64
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.9.1 | linux | x64
gyp info find Python using Python version 3.9.5 found at "/usr/bin/python3"
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/var/[**Redacted**]/Server/node_modules/symbology/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.cache/node-gyp/16.9.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/16.9.1',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/16.9.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/var/[**Redacted**]/Server/node_modules/symbology',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok 
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.9.1 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/var/[**Redacted**]/Server/node_modules/symbology/build'
  CC(target) Release/obj.target/symbology/.zint/backend/output.o
  CC(target) Release/obj.target/symbology/.zint/backend/2of5.o
  CC(target) Release/obj.target/symbology/.zint/backend/auspost.o
  CC(target) Release/obj.target/symbology/.zint/backend/aztec.o
  CC(target) Release/obj.target/symbology/.zint/backend/code.o
  CC(target) Release/obj.target/symbology/.zint/backend/code1.o
  CC(target) Release/obj.target/symbology/.zint/backend/code16k.o
  CC(target) Release/obj.target/symbology/.zint/backend/code49.o
  CC(target) Release/obj.target/symbology/.zint/backend/code128.o
  CC(target) Release/obj.target/symbology/.zint/backend/common.o
  CC(target) Release/obj.target/symbology/.zint/backend/composite.o
  CC(target) Release/obj.target/symbology/.zint/backend/dmatrix.o
  CC(target) Release/obj.target/symbology/.zint/backend/gridmtx.o
  CC(target) Release/obj.target/symbology/.zint/backend/gs1.o
  CC(target) Release/obj.target/symbology/.zint/backend/imail.o
  CC(target) Release/obj.target/symbology/.zint/backend/large.o
  CC(target) Release/obj.target/symbology/.zint/backend/library.o
  CC(target) Release/obj.target/symbology/.zint/backend/maxicode.o
  CC(target) Release/obj.target/symbology/.zint/backend/medical.o
  CC(target) Release/obj.target/symbology/.zint/backend/pdf417.o
  CC(target) Release/obj.target/symbology/.zint/backend/plessey.o
  CC(target) Release/obj.target/symbology/.zint/backend/png.o
  CC(target) Release/obj.target/symbology/.zint/backend/postal.o
  CC(target) Release/obj.target/symbology/.zint/backend/ps.o
../.zint/backend/ps.c: In function 'ps_plot':
../.zint/backend/ps.c:142:9: warning: unused variable 'buf' [-Wunused-variable]
  142 |   char *buf = malloc(sizeof(symbol->rendered_data));
      |         ^~~
  CC(target) Release/obj.target/symbology/.zint/backend/qr.o
  CC(target) Release/obj.target/symbology/.zint/backend/reedsol.o
  CC(target) Release/obj.target/symbology/.zint/backend/rss.o
  CC(target) Release/obj.target/symbology/.zint/backend/svg.o
../.zint/backend/svg.c: In function 'svg_plot':
../.zint/backend/svg.c:125:9: warning: unused variable 'buf' [-Wunused-variable]
  125 |   char *buf = malloc(sizeof(symbol->rendered_data));
      |         ^~~
  CC(target) Release/obj.target/symbology/.zint/backend/telepen.o
  CC(target) Release/obj.target/symbology/.zint/backend/upcean.o
  CC(target) Release/obj.target/symbology/.zint/backend/bmp.o
  CC(target) Release/obj.target/symbology/.zint/backend/codablock.o
  CC(target) Release/obj.target/symbology/.zint/backend/dotcode.o
  CC(target) Release/obj.target/symbology/.zint/backend/eci.o
  CC(target) Release/obj.target/symbology/.zint/backend/emf.o
  CC(target) Release/obj.target/symbology/.zint/backend/gb2312.o
  CC(target) Release/obj.target/symbology/.zint/backend/gb18030.o
  CC(target) Release/obj.target/symbology/.zint/backend/general_field.o
  CC(target) Release/obj.target/symbology/.zint/backend/gif.o
  CC(target) Release/obj.target/symbology/.zint/backend/hanxin.o
  CC(target) Release/obj.target/symbology/.zint/backend/mailmark.o
  CC(target) Release/obj.target/symbology/.zint/backend/pcx.o
  CC(target) Release/obj.target/symbology/.zint/backend/raster.o
  CC(target) Release/obj.target/symbology/.zint/backend/sjis.o
  CC(target) Release/obj.target/symbology/.zint/backend/tif.o
  CC(target) Release/obj.target/symbology/.zint/backend/ultra.o
  CC(target) Release/obj.target/symbology/.zint/backend/vector.o
  CXX(target) Release/obj.target/symbology/src/binding/main.o
../src/binding/main.cpp:10:10: fatal error: nan.h: No such file or directory
   10 | #include <nan.h>
      |          ^~~~~~~
compilation terminated.
make: *** [symbology.target.mk:164: Release/obj.target/symbology/src/binding/main.o] Error 1
make: Leaving directory '/var/[**Redacted**]/Server/node_modules/symbology/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Linux 4.4.0-210-generic
gyp ERR! command "/usr/local/nvm/versions/node/v16.9.1/bin/node" "/usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin/symbology.node" "--module_name=symbology" "--module_path=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v93"
gyp ERR! cwd /var/[**Redacted**]/Server/node_modules/symbology
gyp ERR! node -v v16.9.1
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/nvm/versions/node/v16.9.1/bin/node /usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin/symbology.node --module_name=symbology --module_path=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/var/[**Redacted**]/Server/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:394:28)
node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1064:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
node-pre-gyp ERR! System Linux 4.4.0-210-generic
node-pre-gyp ERR! command "/usr/local/nvm/versions/node/v16.9.1/bin/node" "/var/[**Redacted**]/Server/node_modules/symbology/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /var/[**Redacted**]/Server/node_modules/symbology
node-pre-gyp ERR! node -v v16.9.1
node-pre-gyp ERR! node-pre-gyp -v v1.0.5
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/nvm/versions/node/v16.9.1/bin/node /usr/local/nvm/versions/node/v16.9.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin/symbology.node --module_name=symbology --module_path=/var/[**Redacted**]/Server/node_modules/symbology/bin/binding/bin --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v93' (1)`

--- a/binding.gyp
+++ b/binding.gyp
@@ -60,7 +60,7 @@
         "src/binding/main.cpp"
       ],
       "include_dirs": [
-        "node_modules/nan"
+        "<!(node -e \"require('nan')\")"
       ],
       "msvs_settings": {
         "VCCLCompilerTool": {

